### PR TITLE
test: Cancel block_stream task to workaround infinite lake loop

### PR DESCRIPTION
**Pull request by morgsmccauley**
`near-lake-framework` will infinitely retry, making our tests infinitely hang. This PR monitors the `last_processed_ block`, and cancels the task from the outside when the last desired block is reached.

### CODE REVIEW

File: block-streamer/src/block_stream.rs [hunk 1]
```diff
@@ -479,7 +493,7 @@ mod tests {
 
         mock_lake_s3_client
             .expect_list_common_prefixes()
-            .returning(|_, _| Ok(vec![]));
+            .returning(|_, _| Ok(vec![107503704.to_string(), 107503705.to_string()]));
```
Comment (darunrs) [inline block-streamer/src/block_stream.rs:496]: What caused this to change?

File: block-streamer/src/block_stream.rs [hunk 2]
```diff
@@ -448,23 +452,33 @@ mod tests {
...
+            .start(
+                91940840,
+                std::sync::Arc::new(mock_redis),
+                std::sync::Arc::new(mock_reciever_blocks_processor),
+                mock_lake_s3_client,
+            )
+            .unwrap();
+
+        loop {
+            if last_processed_block_height.load(Ordering::Relaxed) == 107503705 {
```
Comment (darunrs) [inline block-streamer/src/block_stream.rs:472]: Should use >= instead just to be absolutely sure we stop?

**Review by darunrs — APPROVED** [review]
Smart way of solving this issue!

File: block-streamer/src/block_stream.rs (hunk 1 above)
Comment (morgsmccauley) [reply block-streamer/src/block_stream.rs:496]: The test setup wasn't correct, I think they probably got edited while they were `ignored`

File: block-streamer/src/block_stream.rs (hunk 2 above)
Comment (morgsmccauley) [reply block-streamer/src/block_stream.rs:472]: Good point!

**Pull request merged**